### PR TITLE
Always set KUBE_NODE_OS_DISTRIBUTION on gcp

### PR DESCRIPTION
**Pull request by fejta**
/assign @zmerlynn

### CODE REVIEW

File: kubetest/main.go [hunk 1]
```diff
@@ -512,6 +509,11 @@ func prepareGcp(o *options) error {
 	if err := migrateGcpEnvAndOptions(o); err != nil {
 		return err
 	}
+	if o.provider == "gke" && o.gcpNodeImage != "" {
+		// GKE needs woth KUBE_NODE_OS_DISTRIBUTION and KUBE_GKE_IMAGE_TYPE
```
Comment (zmerlynn) [inline kubetest/main.go]: s/woth/both/

File: kubetest/main.go [hunk 2]
```diff
@@ -495,7 +492,7 @@ func migrateGcpEnvAndOptions(o *options) error {
 			name:   "--gcp-network",
 		},
 		{
-			env:    nodeImage,
+			env:    "KUBE_NODE_OS_DISTRIBUTION",
 			option: &o.gcpNodeImage,
```
Comment (zmerlynn) [inline kubetest/main.go]: What's the resolution order if o.gcpNodeImage is set *and* KUBE_NODE_OS_DISTRIBUTION is set?

I think some of the migration I did has both set now.

Comment (fejta) [inline kubetest/main.go]: I believe it leaves them alone? Let's head over and consult the unit test :grin:

Comment (fejta) [inline kubetest/main.go]: Nope, looks like the flag clobbers the env: https://github.com/kubernetes/test-infra/blob/master/kubetest/util_test.go#L374

Comment (fejta) [inline kubetest/main.go]: Are there reasons we want these to differ?

Comment (zmerlynn) [inline kubetest/main.go]: Not that I can think of. I believe gcloud will take `gci` as well, but I'll confirm in just a bit if I get a change.

File: kubetest/main.go (hunk 1 above)
Comment (fejta) [inline kubetest/main.go]: Done

File: kubetest/main.go [hunk 3]
```diff
@@ -512,6 +509,11 @@ func prepareGcp(o *options) error {
 	if err := migrateGcpEnvAndOptions(o); err != nil {
 		return err
 	}
+	if o.provider == "gke" && o.gcpNodeImage != "" {
+		// GKE needs both KUBE_NODE_OS_DISTRIBUTION and KUBE_GKE_IMAGE_TYPE
```
Comment (zmerlynn) [inline kubetest/main.go]: Looking at this again, it looks like both GCE and GKE need `NODE_OS_DISTRIBUTION` set. `KUBE_NODE_OS_DISTRUBTION` will get into `NODE_OS_DISTRIBUTION` if it's run through bash, but the new GKE deployment doesn't handle it, for instance.

**Comment by fejta** [comment]
Looks like I can close my pr, right?

**Pull request closed without merging**